widget/accordioncontainer.go

```diff
@@ -0,0 +1,335 @@
+package widget
+
+import (
+	"image/color"
+
+	"fyne.io/fyne"
+	"fyne.io/fyne/canvas"
+	"fyne.io/fyne/driver/desktop"
+	"fyne.io/fyne/internal/cache"
+	"fyne.io/fyne/theme"
+)
+
+// AccordionContainer displays a list of AccordionItems.
+// Each item is represented by a button that reveals a detailed view when tapped.
+type AccordionContainer struct {
+	BaseWidget
+
+	items []*accordionItem
+}
+
+// NewAccordionContainer creates a new accordion widget.
+func NewAccordionContainer() *AccordionContainer {
+	a := &AccordionContainer{
+		BaseWidget: BaseWidget{},
+	}
+	a.ExtendBaseWidget(a)
+	return a
+}
+
+// MinSize returns the size that this widget should not shrink below.
+func (a *AccordionContainer) MinSize() fyne.Size {
+	a.ExtendBaseWidget(a)
+	return a.BaseWidget.MinSize()
+}
+
+// Append adds the given header and detail view to this AccordionContainer.
+func (a *AccordionContainer) Append(header string, detail fyne.CanvasObject) {
+	item := &accordionItem{
+		BaseWidget: BaseWidget{},
+		container:  a,
+		detail:     detail,
+		open:       true,
+	}
+	item.header = &accordionItemHeader{
+		BaseWidget: BaseWidget{},
+		item:       item,
+		text:       header,
+	}
+	item.header.ExtendBaseWidget(item.header)
+	item.ExtendBaseWidget(item)
+
+	r := cache.Renderer(a).(*accordionContainerRenderer)
+	a.items = append(a.items, item)
+	r.objects = append(r.objects, item)
+	a.Refresh()
+}
+
+// Remove deletes the item at the given index from this AccordionContainer.
+func (a *AccordionContainer) Remove(index int) {
+	r := cache.Renderer(a).(*accordionContainerRenderer)
+	a.items = append(a.items[:index], a.items[index+1:]...)
+	r.objects = append(r.objects[:index], r.objects[index+1:]...)
+	a.Refresh()
+}
+
+// Open expands the item at the given index.
+func (a *AccordionContainer) Open(index int) {
+	if index < 0 || index >= len(a.items) {
+		return
+	}
+	a.items[index].setOpen(true)
+	a.Refresh()
+}
+
+// OpenAll expands all items.
+func (a *AccordionContainer) OpenAll() {
+	for _, i := range a.items {
+		i.setOpen(true)
+	}
+	a.Refresh()
+}
+
+// Close collapses the item at the given index.
+func (a *AccordionContainer) Close(index int) {
+	if index < 0 || index >= len(a.items) {
+		return
+	}
+	a.items[index].setOpen(false)
+	a.Refresh()
+}
+
+// CloseAll collapses all items.
+func (a *AccordionContainer) CloseAll() {
+	for _, i := range a.items {
+		i.setOpen(false)
+	}
+	a.Refresh()
+}
+
+// CreateRenderer is a private method to Fyne which links this widget to its renderer
+func (a *AccordionContainer) CreateRenderer() fyne.WidgetRenderer {
+	var os []fyne.CanvasObject
+	for _, i := range a.items {
+		os = append(os, i)
+	}
+	return &accordionContainerRenderer{
```

Comment:
Having so little in the renderer creation seems strange - and may be the cause of the inner objects needing references up to the parent etc.
Perhaps this container should be responsible for rendering all the headers and laying out the content? There is actually no need for the item to be a widget at all I think... just like TabItem in TabController.

Reply:
There's still nothing in renderer creation but it now calls the same auxiliary called by Refresh which creates the buttons as needed